packages/next/src/client/legacy/image.tsx

```diff
@@ -865,8 +865,8 @@ export default function Image({
           (config.localPatterns.length === 1 &&
             config.localPatterns[0].pathname === '/_next/static/media/**'))
       ) {
-        warnOnce(
-          `Image with src "${src}" is using a query string which is not configured in images.localPatterns. This config will be required starting in Next.js 16.` +
+        throw new Error(
+          `Image with src "${src}" is using a query string which requires images.localPatterns configuration.` +
```

Comment:
This one will also need to be moved (I think legacy might use a different loader)